### DISCUSSION TITLE
How to debug the MQTT connection with wireguard?

### DISCUSSION THREAD

**Discussion by skirsten** [greghesp/ha-bambulab#391] [General]
Hi, just a general question: I want to compare the traffic of pybambu and BambuStudio to understand how it works and what is being sent etc.

I tried to use `SSLKEYLOGFILE` (https://everything.curl.dev/usingcurl/tls/sslkeylogfile, https://wiki.wireshark.org/TLS#using-the-pre-master-secret) on the BambuStudio and it allows me to see the decrypted HTTPS messages for the feed etc. in wireshark.

But I cannot get wireshark to decrypt the TLS of the MQTT messages :disappointed: 

So just wanted to ask how you figured out the message format etc. or how you are going about debugging the communication between the bambu network plugin and the printer.

**Comment by jneilliii** [comment]
I use MQTT Explorer and connect to the X1 directly. 

![image](https://github.com/greghesp/ha-bambulab/assets/5249455/490a585c-56aa-4c32-b163-4d12bd5c89ac)

anything that comes in on the report topic is status info anything on the request topic is commands sent from slicer.

they will be found under device/[serial number]/report and device/[serial number]/request

another resource: https://github.com/Doridian/OpenBambuAPI/blob/main/mqtt.md

**Comment by skirsten** [comment]
Thanks a lot. I was able to connect to the printer with the MQTT Explorer. Unfortunately it seems my A1 Mini only allows one connection or something like this because its throwing me out of the BambuStudio shortly after.

I also tried to connect via ftps but I haven't figured it out yet (always getting a timeout).

I was hoping to just build a small MQTT proxy/forwarder that BambuStudio and the explorer can connect to at the same time, but its not possible to manually add a printer via IP to BambuStudio :face_exhaling:.

Will now try mitmproxy or https://github.com/nikitastupin/mitmproxy-mqtt-script lets see.

But thanks for your help!
